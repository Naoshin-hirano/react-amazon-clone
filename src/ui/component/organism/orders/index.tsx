import React from "react";
import "../../../../design/component/atoms/orders/index.css";
import { Order } from "./order";

export const Orders: React.FC<any> = (props) => {
    const { orders } = props;
    return (
        <div className="orders">
            <h1> Your Orders</h1>
            <div className="orders__order">
                {orders?.map((order: any, key: any) => (
                    <Order order={order} key={key} />
                ))}
            </div>
        </div>
    );
};
