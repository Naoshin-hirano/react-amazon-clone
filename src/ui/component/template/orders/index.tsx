import React from "react";
import { Header } from "ui/component/organism/common/header";
import { Orders } from "ui/component/organism/orders";

export const Template: React.FC<any> = (props) => {
    return (
        <div>
            <Header />
            <Orders {...props} />
        </div>
    );
};
