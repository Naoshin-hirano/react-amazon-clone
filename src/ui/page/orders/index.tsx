import React from "react";
import { Template } from "ui/component/template/orders";
import { useCreateProps } from "./presentation";

export const OrdersPage: React.FC = () => {
    const props = useCreateProps();
    return <Template {...props}></Template>;
};
