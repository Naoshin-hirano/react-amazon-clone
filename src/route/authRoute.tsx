import { Route } from "@tanstack/react-location";
import React from "react";
import { LoginPage } from "ui/page/login";
import { RegisterPage } from "ui/page/register";

export const authRoute: Route[] = [
    {
        path: "/login",
        element: <LoginPage />,
    },
    {
        path: "/register",
        element: <RegisterPage />,
    },
];
