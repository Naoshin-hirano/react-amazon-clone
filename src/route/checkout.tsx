import { Route } from "@tanstack/react-location";
import React from "react";
// import { SocondSamplePage } from "ui/page/second";
import { CheckoutPage } from "ui/page/checkout";

export const checkoutRoute: Route[] = [
    {
        path: "/checkout",
        element: <CheckoutPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
