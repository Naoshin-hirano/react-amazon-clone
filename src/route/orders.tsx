import { Route } from "@tanstack/react-location";
import React from "react";
import { OrdersPage } from "ui/page/orders";

export const ordersRoute: Route[] = [
    {
        path: "/orders",
        element: <OrdersPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
