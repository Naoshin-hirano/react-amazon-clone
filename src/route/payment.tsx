import { Route } from "@tanstack/react-location";
import React from "react";
import { PaymentPage } from "ui/page/payment";

export const paymentRoute: Route[] = [
    {
        path: "/payment",
        element: <PaymentPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
