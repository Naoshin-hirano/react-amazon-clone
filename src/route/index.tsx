import { ReactLocation, Route } from "@tanstack/react-location";
import { history } from "./history";
import { authRoute } from "route/authRoute";
import React from "react";
import { TopPage } from "ui/page/top";

const indexRoute: Route[] = [
    {
        path: "/",
        element: <TopPage />,
    },
    {
        path: "/top",
        element: <TopPage />,
    },
];

export const routes: Route[] = [authRoute].reduce(
    (prevRoute, currRoute) => prevRoute.concat(currRoute),
    indexRoute
);

export const location = new ReactLocation({ history: history });
