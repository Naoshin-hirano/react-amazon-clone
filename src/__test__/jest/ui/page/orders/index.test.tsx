import { render } from "@testing-library/react";
import React from "react";
import { Router } from "@tanstack/react-location";
import { location, routes } from "../../../../../route";
import { OrdersPage } from "ui/page/orders";

describe("OrdersPage", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("TopPage画面 snapshot確認", async () => {
        const { asFragment } = render(
            <Router routes={routes} location={location}>
                <OrdersPage />
            </Router>
        );
        expect(asFragment()).toMatchSnapshot("OrdersPage");
    });
});
