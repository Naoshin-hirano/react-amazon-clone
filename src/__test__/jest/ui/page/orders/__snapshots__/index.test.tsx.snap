// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersPage TopPageÁîªÈù¢ snapshotÁ¢∫Ë™ç: OrdersPage 1`] = `
<DocumentFragment>
  <div>
    <div
      class="header"
    >
      <a
        href="/"
      >
        <img
          alt="Amazon logo"
          class="header__logo"
          src="http://pngimg.com/uploads/amazon/amazon_PNG11.png"
        />
      </a>
      <div
        class="header__search"
      >
        <input
          class="header__searchInput"
          type="text"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root header__searchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <div
        class="header__nav"
      >
        <a
          href="/login"
        >
          <div
            class="header__option"
          >
            <span
              class="header__optionLineOne"
            >
              Hello xxxx@yyyy.com
            </span>
            <span
              class="header__optionLineTwo"
            >
              Sign In
            </span>
          </div>
        </a>
        <a
          href="/orders"
        >
          <div
            class="header__option"
          >
            <span
              class="header__optionLineOne"
            >
              Returns
            </span>
            <span
              class="header__optionLineTwo"
            >
              & Orders
            </span>
          </div>
        </a>
        <div
          class="header__option"
        >
          <span
            class="header__optionLineOne"
          >
            Your
          </span>
          <span
            class="header__optionLineTwo"
          >
            Prime
          </span>
        </div>
        <a
          href="/checkout"
        >
          <div
            class="header__optionBasket"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.21 9l-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
              />
            </svg>
            <span
              class="header__optionLineTwo header__basketCount"
            >
              3
            </span>
          </div>
        </a>
      </div>
    </div>
    <div
      class="orders"
    >
      <h1>
         Your Orders
      </h1>
      <div
        class="orders__order"
      >
        <div
          class="order"
        >
          <h2>
            Order
          </h2>
          <p>
            October 21st 2022, 10:46pm
          </p>
          <p
            class="order__id"
          >
            <small>
              Order ID: kgkaeethavafaffa
            </small>
          </p>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://www.anoyama.com/wp-content/uploads/2019/05/0_equipment-860x550.jpg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                ÁôªÂ±±„Ç∞„ÉÉ„Ç∫‰∏ÄÂºè
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  23,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
              </div>
            </div>
          </div>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://images.camphack.jp/wp-content/uploads/2022/02/image19.jpeg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                „Ç≠„É£„É≥„Éó„Ç∞„ÉÉ„Ç∫‰∏ÄÂºè
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  53,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
              </div>
            </div>
          </div>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://m.media-amazon.com/images/I/616AAPLKJTL._AC_UY500_.jpg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                ÁôªÂ±±„Ç∑„É•„Éº„Ç∫
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  16,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
              </div>
            </div>
          </div>
          <h3
            class="order__total"
          >
            Order Total:$666
          </h3>
        </div>
        <div
          class="order"
        >
          <h2>
            Order
          </h2>
          <p>
            October 17th 2022, 3:46pm
          </p>
          <p
            class="order__id"
          >
            <small>
              Order ID: yyyhaearearegfaeare
            </small>
          </p>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://www.anoyama.com/wp-content/uploads/2019/05/0_equipment-860x550.jpg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                ÁôªÂ±±„Ç∞„ÉÉ„Ç∫‰∏ÄÂºè
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  23,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
              </div>
            </div>
          </div>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://images.camphack.jp/wp-content/uploads/2022/02/image19.jpeg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                „Ç≠„É£„É≥„Éó„Ç∞„ÉÉ„Ç∫‰∏ÄÂºè
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  53,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
              </div>
            </div>
          </div>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://m.media-amazon.com/images/I/616AAPLKJTL._AC_UY500_.jpg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                ÁôªÂ±±„Ç∑„É•„Éº„Ç∫
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  16,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
              </div>
            </div>
          </div>
          <h3
            class="order__total"
          >
            Order Total:$246
          </h3>
        </div>
        <div
          class="order"
        >
          <h2>
            Order
          </h2>
          <p>
            October 19th 2022, 5:00pm
          </p>
          <p
            class="order__id"
          >
            <small>
              Order ID: waeimganreirangaef
            </small>
          </p>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://www.anoyama.com/wp-content/uploads/2019/05/0_equipment-860x550.jpg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                ÁôªÂ±±„Ç∞„ÉÉ„Ç∫‰∏ÄÂºè
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  23,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
              </div>
            </div>
          </div>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://images.camphack.jp/wp-content/uploads/2022/02/image19.jpeg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                „Ç≠„É£„É≥„Éó„Ç∞„ÉÉ„Ç∫‰∏ÄÂºè
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  53,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
              </div>
            </div>
          </div>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://m.media-amazon.com/images/I/616AAPLKJTL._AC_UY500_.jpg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                ÁôªÂ±±„Ç∑„É•„Éº„Ç∫
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  16,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
                <p>
                  üåü
                </p>
              </div>
            </div>
          </div>
          <h3
            class="order__total"
          >
            Order Total:$578
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
